AngularJS/src/org/angular2/entities/source/Angular2SourceDirective.java

```diff
@@ -143,6 +161,51 @@ public boolean isTemplate() {
                 Collections.unmodifiableCollection(outputs.values()));
   }
 
+  @NotNull
+  private Angular2CtorBoundParameters getCtorParameters() {
```

Comment:
I am not sure, but I think you should traverse the class hierarchy as the `@Attribute` could be specified in parent class.

Reply:
But Angular needs to know how to inject the constructor parameter, or it will emit a `NullInjectorError`. I did some experiments and the only constructor which seems to matter is the sub-class one, if there is a hierarchy.

Reply:
Ok, great.